selfhost/parser.jakt

```diff
@@ -291,6 +259,38 @@ boxed enum ParsedStatement {
     }
 }
 
+struct ParsedBlock {
```

Comment:
does this need to move because of what you're doing in the PR?

Reply:
Yes, previously `typecheck_enum` on enums was called before `typecheck_struct` on structs. As they are in the same loop now the ordering for type checking methods in enums/structs becomes important.

Reply:
Hmm, the language is supposed to allow arbitrary declaration order. If the ordering is significant, then I think something has gone wrong :)

Reply:
Ah, yes that does make sense :)
The second commit here for enum function return type actually fixed the ordering issue, I have removed the reorder.